doc/api/net.md

```diff
@@ -75,6 +75,16 @@ added: v15.0.0
 
 Adds a rule to block the given IP address.
 
+### `blockList.removeAddress(address[, type])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `address` {string|net.SocketAddress} An IPv4 or IPv6 address.
+* `type` {string} Either `'ipv4'` or `'ipv6'`. **Default:** `'ipv4'`.
+
+Removes a rule to block the given IP address.
```

Comment:
The documentation for these functions should specify whether they throw if no such rule was found, and they should probably clarify that, based on my understanding, they require exact matches. For example, it is not possible to remove a part of an existing range rule.

Reply:
@tniessen Good point, the removal methods don't currently throw if a matching rule isn't found. I think that behavior makes sense. I can implement / document it if that's alright?

Reply:
hmm or the method could return a boolean indicating whether a matching rule was removed?

Reply:
I'm not sure what the behavior of the adding functions is (I'm on mobile right now so can't test).

- If add throws if a rule already exists, remove should throw if no such rule exists.
- If add doesn't add duplicate rules, maybe this should return a boolean (similar to JS sets).
- If add does add duplicate rules, and if this removes exactly one of them (as opposed to all of them), then removal should throw if no rule exists that could be removed.

Reply:
Gotcha, so add *used to* allow duplicate rules. I changed this behavior to ease rule removal, so add *no longer* allows duplicate rules. Maybe this change is undesirable. I could be missing context on why duplicate rules were allowed in the first place.

Otherwise, I agree with your points. Remove probably shouldn't throw since add doesn't. It should return a boolean like map, set, etc. as you mentioned.